src/framebuffer.rs

```diff
@@ -178,18 +189,30 @@ const fn shift_msb0(input: usize) -> usize {
 }
 
 macro_rules! impl_bit {
-    (@impl $raw_storage_type:ty, $raw_type:ty, $shifter:expr) => {
+    (@impl $raw_storage_type:ty, $raw_type:ty, $shifter:expr, $dir:ty) => {
         impl<CS, const WIDTH: usize, const HEIGHT: usize, const N: usize> SetPixelRaw<$raw_storage_type>
-            for Framebuffer<CS, WIDTH, HEIGHT, N>
+            for Framebuffer<CS, $dir, WIDTH, HEIGHT, N>
         {
             fn set_pixel_raw(&mut self, p: Point, r: $raw_type) {
                 if let (Ok(x), Ok(y)) = (usize::try_from(p.x), usize::try_from(p.y)) {
                     if x < WIDTH && y < HEIGHT {
-                        let pixels_per_bit = 8 / <$raw_type>::BITS_PER_PIXEL;
-                        let bits_per_row = WIDTH * <$raw_type>::BITS_PER_PIXEL;
-                        let bytes_per_row = (bits_per_row + 7) / 8;
-                        let byte_index = bytes_per_row * y + (x / pixels_per_bit);
-                        let bit_index = 8 - (x % pixels_per_bit + 1) * <$raw_type>::BITS_PER_PIXEL;
+                      let (bit_index, byte_index) =  if <$dir>::IS_HORIZONTAL {
+                            let pixels_per_bit = 8 / <$raw_type>::BITS_PER_PIXEL;
+                            let bits_per_row = WIDTH * <$raw_type>::BITS_PER_PIXEL;
+                            let bytes_per_row = (bits_per_row + 7) / 8;
+                            let byte_index = bytes_per_row * y + (x / pixels_per_bit);
+                            let bit_index = 8 - (x % pixels_per_bit + 1) * <$raw_type>::BITS_PER_PIXEL;
+
+                            (bit_index, byte_index)
+                        } else {
+                            let y_start_byte = (y * <$raw_type>::BITS_PER_PIXEL) / 8;
+                            let y_start_bit = (y * <$raw_type>::BITS_PER_PIXEL) % 8;
+
+                            let byte_index = WIDTH * y_start_byte + x;
+
+
+                            (y_start_bit, byte_index)
```

Comment:
Shouldn't this use the same logic as the horizontal case? It wouldn't be compatible with `ImageRaw` if the number of bytes per column isn't an integer.